plugins/jamf/custom_actions.go

```diff
@@ -0,0 +1,109 @@
+package jamf
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/blinkops/blink-http/consts"
+	"github.com/blinkops/blink-http/implementation/requests"
+	"github.com/blinkops/blink-http/plugins/types"
+	"github.com/blinkops/blink-sdk/plugin"
+	"net/http"
+	"strconv"
+	"strings"
+)
+
+const (
+	deviceSerialsParam = "Device Serials"
+	listFilter         = "Return only active devices"
+	pluginName         = "jamf"
+)
+
+func getInstalledDevices(ctx *plugin.ActionContext, request *plugin.ExecuteActionRequest, plugin types.Plugin) ([]byte, error) {
+	inputDeviceSerials, returnInstalledDevices, err := getGetInstalledDevicesParams(request)
+	if err != nil {
+		return nil, err
+	}
+
+	installedDeviceSerials, err := performGetInstalledDevices(ctx, plugin, request, inputDeviceSerials, returnInstalledDevices)
+	if err != nil {
+		return nil, err
+	}
+
+	if installedDeviceSerials == nil {
+		installedDeviceSerials = []string{}
+	}
+	activeSerialsStr, err := json.Marshal(installedDeviceSerials)
+	if err != nil {
+		return nil, errors.New("failed to marshal installed device serials json")
+	}
+
+	return activeSerialsStr, nil
+}
+
+func getGetInstalledDevicesParams(request *plugin.ExecuteActionRequest) ([]string, bool, error) {
+	params, err := request.GetParameters()
+	if err != nil {
+		return nil, false, err
+	}
+
+	deviceSerials, returnOnlyActiveDevices := params[deviceSerialsParam], params[listFilter]
+	if deviceSerials == "" {
+		return nil, false, errors.New("no device serials provided")
+	}
+
+	returnInstalledDevicesBool, err := strconv.ParseBool(returnOnlyActiveDevices)
+	if err != nil {
+		return nil, false, errors.New("unable to convert 'return active devices only' to boolean")
+	}
+
+	deviceSerials = strings.ReplaceAll(deviceSerials, ", ", ",")
+	deviceSerialsList := strings.Split(deviceSerials, ",")
+
+	return deviceSerialsList, returnInstalledDevicesBool, nil
+}
+
+func performGetInstalledDevices(ctx *plugin.ActionContext, plugin types.Plugin, request *plugin.ExecuteActionRequest, inputDeviceSerials []string, returnOnlyActiveDevices bool) ([]string, error) {
+	requestUrl, err := requests.GetRequestUrl(ctx, pluginName)
+	if err != nil {
+		return nil, errors.New(consts.RequestUrlMissing)
+	}
+
+	installedDevices, err := getDeviceSerialsAndStatus(ctx, plugin, requestUrl, request.Timeout)
+	if err != nil {
+		return nil, err
+	}
+
+	var installedSerials []string
```

Comment:
initialize the slice so it can't be nil
```suggestion
	installedSerials := []string{}
```